cocotb/README.md

```diff
@@ -0,0 +1,312 @@
+# COCOTB (Coroutine Based Cosimulation Test Bench)
+
+Doğrulama, debug, test... FPGA tasarımında sadece ve sadece RTL kod yazıp çalışmasını beklemek çok zor. Yaptığımız projeler gereği tek seferde bir kodu çalıştırmamız imkansız değil fakat corner-case diyebileceğimiz koşullarda kodun çalışmasını sağlamak ayrı bir zorluk.
+
+Bunu başarabilmek için kart üzerinde gerçek zamanlı testler öncesi yapacağımız adımlar çok önemli: Kodu doğrulamak ve recursive olarak kodu düzeltmek. Buna da aslında RTL doğrulama diyebiliriz.
+
+Doğrulama konusunda UVM gibi çokça kullanılan bir yöntem var ama bu yazıda Python temelli `cocotb`'den bahsedeceğim. Evet, verilog, system-verilog veya VHDL ile yazılan bir doğrulamadan bahsetmiyorum. Tamamiyle Python kullanarak RTL tasarımınızı özgürce doğrulayabiliyorsunuz. Ayrıca kullanımı çok kolay.
+
+Yazının devamında `cocotb` konusunda genel bilgilerden bahsettim. Sonrasında Windows bilgisayarıma nasıl kurduğumdan bahsettim. En son da örnek birkaç kullanım gösterdim.
+
+Cocotb konusunda ileride örnek projeler üzerinden gösterimler yapmayı planlıyorum. Bu yazıyı bir giriş ve tanıtım olarak görebilirsiniz.
+
+## Neden COCOTB?
+
+Bu soruya verilebilecek birçok cevap var fakat en önemlilerini aşağıda
+sıraladım:
+
+- Python dili temel manada kullanım açısından öğrenmesi ve yazması kolay bir dil.
+- `cocotb` ileri düzeyde Python bilgisi gerektirmiyor.
+- Python'ın sağladığı çok geniş bir kütüphane içeriği var (numpy, pandas...
+  gibi). RTL tabanlı bir doğrulama aracında kolayca yapamayacağınız işlemleri
+  (rastgele sayı üretme, array yapıları, ethernet paketleri... vs.) Python'da çok kolay bir şekilde
+  yapabilirsiniz.
+- Jenkins entegrasyonu sağlaması adına built-in desteğine sahip yani git tabanlı
```

Comment:
Jenkins şart değil, komut satırından çalıştırabildiğin için Gitlab CI'da da çalışır.